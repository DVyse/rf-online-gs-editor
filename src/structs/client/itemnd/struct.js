import Struct from '../../../classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: String, name: 'strName', len: 64 } },
]);
