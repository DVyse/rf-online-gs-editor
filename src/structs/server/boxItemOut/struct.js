import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  {
    child: [
      { type: String, len: 8, name: 'strItemCode' },
      { type: Number, len: 32, name: 'nItemCount' },
      { type: Number, len: 32, name: 'nItemProb' },
    ],
    repeat: 61,
  },
]);
