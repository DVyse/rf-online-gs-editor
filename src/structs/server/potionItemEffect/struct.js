import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  { child: { type: Number, name: 'nClass', len: 32 } },
  { child: { type: Number, name: 'nIconIDX', len: 32 } },
  { child: { type: Number, name: 'nMastIndex', len: 32 } },
  { child: { type: String, name: 'strMastKorName', len: 64 } },
  { child: { type: String, name: 'strMastEngName', len: 64 } },
  { child: { type: String, name: 'strKorName', len: 64 } },
  { child: { type: String, name: 'strEngName', len: 64 } },
  { child: { type: Number, name: 'nLv', len: 32 } },
  { child: { type: Boolean, name: 'bActivate', len: 32 } },
  { child: { type: Boolean, name: 'bEnable', len: 32 } },
  { child: { type: String, name: 'strUsableRace', len: 64 } },
  { child: { type: String, name: 'strActableDst', len: 64 } },
  { child: { type: String, name: 'strGradeLimit', len: 64 } },
  { child: { type: Number, name: 'nNeedMastIndex', len: 32 } },
  { child: { type: String, name: 'strFixWeapon', len: 64 } },
  { child: { type: Boolean, name: 'bFixshield', len: 32 } },
  { child: { type: Number, name: 'nSpecialType', len: 32 } },
  { child: { type: Number, name: 'nNeedSpecialType', len: 32 } },
  { child: { type: Number, name: 'nNeedHP', len: 32 } },
  { child: { type: Number, name: 'nNeedFP', len: 32 } },
  { child: { type: Number, name: 'nNeedSP', len: 32 } },

  // _consume_item_list
  {
    child: [
      {
        type: String,
        name: 'itmNeedItemCode',
        len: 8,
      },
      { type: Number, name: 'nNeedItemCount', len: 32 },
    ],
    repeat: 3,
  },

  { child: { type: Number, name: 'fActDelay', len: 32, as: 'float' } },
  { child: { type: Boolean, name: 'bCumulType', len: 32 } },
  { child: { type: Number, name: 'nCumulCounter', len: 32 } },
  { child: { type: Number, name: 'nNewEffCount', len: 32 } },
  { child: { type: String, name: 'strEffectCode', len: 64 } },
  { child: { type: Number, name: 'nAttackable', len: 32 } },
  { child: { type: Number, name: 'nAttType', len: 32 }, repeat: 7 },
  { child: { type: Number, name: 'nAttConstant', len: 32 }, repeat: 7 },
  {
    child: { type: Number, name: 'fAttFormulaConstant', len: 32, as: 'float' },
  },
  { child: { type: Number, name: 'nAttNeedBt', len: 32 } },
  { child: { type: Number, name: 'nBonusDistance', len: 32 } },
  { child: { type: String, name: 'strRangeEffCode', len: 64 } },
  { child: { type: Number, name: 'nTempEffectType', len: 32 } },
  { child: { type: Number, name: 'nTempParamCode', len: 32 } },
  {
    child: { type: Number, name: 'fTempValue', len: 32, as: 'float' },
    repeat: 7,
  },
  { child: { type: Number, name: 'nContEffectType', len: 32 } },
  { child: { type: Number, name: 'nEffLimType', len: 32 } },
  { child: { type: Number, name: 'nEffLimType2', len: 32 } },
  { child: { type: Number, name: 'nContAreaType', len: 32 } },

  // _cont_param_list
  {
    child: [
      { type: Number, name: 'nContParamCode', len: 32 },
      { type: Number, name: 'nContParamIndex', len: 32 },
      {
        child: { type: Number, name: 'fContValue', len: 32, as: 'float' },
        repeat: 7,
      },
    ],
    repeat: 5,
  },

  { child: { type: Number, name: 'nContEffectSec', len: 32 }, repeat: 7 },
  { child: { type: Number, name: 'nEtc', len: 32 } },
  { child: { type: Number, name: 'f1s2speed', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f1s2distance', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f2s3speed', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f2s3distance', len: 32, as: 'float' } },
  { child: { type: Number, name: 'nEffectClass', len: 32 } },
]);
