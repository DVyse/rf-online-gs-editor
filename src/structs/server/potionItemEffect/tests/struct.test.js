import struct from '../struct';

describe('Structs', () => {
  describe('Server', () => {
    describe('PotionItemEffect', () => {
      describe('Struct', () => {
        it('should exist', () => {
          expect(struct.toPlainObject()).toMatchSnapshot();
        });
      });
    });
  });
});
