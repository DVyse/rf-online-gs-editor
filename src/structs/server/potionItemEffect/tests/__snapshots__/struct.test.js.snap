// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structs Server PotionItemEffect Struct should exist 1`] = `
Object {
  "fieldNames": Array [
    "nIndex",
    "strCode",
    "nClass",
    "nIconIDX",
    "nMastIndex",
    "strMastKorName",
    "strMastEngName",
    "strKorName",
    "strEngName",
    "nLv",
    "bActivate",
    "bEnable",
    "strUsableRace",
    "strActableDst",
    "strGradeLimit",
    "nNeedMastIndex",
    "strFixWeapon",
    "bFixshield",
    "nSpecialType",
    "nNeedSpecialType",
    "nNeedHP",
    "nNeedFP",
    "nNeedSP",
    "itmNeedItemCode__1_1",
    "nNeedItemCount__1_2",
    "itmNeedItemCode__2_1",
    "nNeedItemCount__2_2",
    "itmNeedItemCode__3_1",
    "nNeedItemCount__3_2",
    "fActDelay",
    "bCumulType",
    "nCumulCounter",
    "nNewEffCount",
    "strEffectCode",
    "nAttackable",
    "nAttType__1",
    "nAttType__2",
    "nAttType__3",
    "nAttType__4",
    "nAttType__5",
    "nAttType__6",
    "nAttType__7",
    "nAttConstant__1",
    "nAttConstant__2",
    "nAttConstant__3",
    "nAttConstant__4",
    "nAttConstant__5",
    "nAttConstant__6",
    "nAttConstant__7",
    "fAttFormulaConstant",
    "nAttNeedBt",
    "nBonusDistance",
    "strRangeEffCode",
    "nTempEffectType",
    "nTempParamCode",
    "fTempValue__1",
    "fTempValue__2",
    "fTempValue__3",
    "fTempValue__4",
    "fTempValue__5",
    "fTempValue__6",
    "fTempValue__7",
    "nContEffectType",
    "nEffLimType",
    "nEffLimType2",
    "nContAreaType",
    "nContParamCode__1_1",
    "nContParamIndex__1_2",
    "fContValue__1_3_1",
    "fContValue__1_3_2",
    "fContValue__1_3_3",
    "fContValue__1_3_4",
    "fContValue__1_3_5",
    "fContValue__1_3_6",
    "fContValue__1_3_7",
    "nContParamCode__2_1",
    "nContParamIndex__2_2",
    "fContValue__2_3_1",
    "fContValue__2_3_2",
    "fContValue__2_3_3",
    "fContValue__2_3_4",
    "fContValue__2_3_5",
    "fContValue__2_3_6",
    "fContValue__2_3_7",
    "nContParamCode__3_1",
    "nContParamIndex__3_2",
    "fContValue__3_3_1",
    "fContValue__3_3_2",
    "fContValue__3_3_3",
    "fContValue__3_3_4",
    "fContValue__3_3_5",
    "fContValue__3_3_6",
    "fContValue__3_3_7",
    "nContParamCode__4_1",
    "nContParamIndex__4_2",
    "fContValue__4_3_1",
    "fContValue__4_3_2",
    "fContValue__4_3_3",
    "fContValue__4_3_4",
    "fContValue__4_3_5",
    "fContValue__4_3_6",
    "fContValue__4_3_7",
    "nContParamCode__5_1",
    "nContParamIndex__5_2",
    "fContValue__5_3_1",
    "fContValue__5_3_2",
    "fContValue__5_3_3",
    "fContValue__5_3_4",
    "fContValue__5_3_5",
    "fContValue__5_3_6",
    "fContValue__5_3_7",
    "nContEffectSec__1",
    "nContEffectSec__2",
    "nContEffectSec__3",
    "nContEffectSec__4",
    "nContEffectSec__5",
    "nContEffectSec__6",
    "nContEffectSec__7",
    "nEtc",
    "f1s2speed",
    "f1s2distance",
    "f2s3speed",
    "f2s3distance",
    "nEffectClass",
  ],
  "fields": Array [
    Object {
      "as": undefined,
      "len": 32,
      "name": "nIndex",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strCode",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nClass",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nIconIDX",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nMastIndex",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strMastKorName",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strMastEngName",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strKorName",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strEngName",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nLv",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bActivate",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bEnable",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strUsableRace",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strActableDst",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strGradeLimit",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedMastIndex",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strFixWeapon",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bFixshield",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nSpecialType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedSpecialType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedHP",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedFP",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedSP",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 8,
      "name": "itmNeedItemCode__1_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 8,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedItemCount__1_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 8,
      "name": "itmNeedItemCode__2_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 8,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedItemCount__2_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 8,
      "name": "itmNeedItemCode__3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 8,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNeedItemCount__3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fActDelay",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bCumulType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nCumulCounter",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nNewEffCount",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strEffectCode",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttackable",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttType__7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttConstant__7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttFormulaConstant",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAttNeedBt",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nBonusDistance",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strRangeEffCode",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nTempEffectType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nTempParamCode",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTempValue__7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nEffLimType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nEffLimType2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContAreaType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamCode__1_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamIndex__1_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__1_3_7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamCode__2_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamIndex__2_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__2_3_7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamCode__3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamIndex__3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__3_3_7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamCode__4_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamIndex__4_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__4_3_7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamCode__5_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContParamIndex__5_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fContValue__5_3_7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__4",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__5",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__6",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nContEffectSec__7",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nEtc",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "f1s2speed",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "f1s2distance",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "f2s3speed",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "f2s3distance",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nEffectClass",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
  ],
  "isDynamic": false,
  "weight": 1168,
}
`;
